pandas/core/resample.py

```diff
@@ -1797,3 +1795,28 @@ def asfreq(obj, freq, method=None, how=None, normalize=False, fill_value=None):
             new_obj.index = new_obj.index.normalize()
 
     return new_obj
+
+
+def _asfreq_compat(index, freq):
+    """
```

Comment:
should this be in the indexes sub-tree?

Reply:
if we start using it elsewhere then we can move it.  as long as its just here, id prefer to keep it private